packages/server/test/support/fixtures/projects/e2e/cypress/integration/screenshots_spec.js

```diff
@@ -302,13 +302,28 @@ describe('taking screenshots', () => {
     })
   })
 
+  it('can pass overwrite option to replace existing filename', () => {
+    cy.viewport(600, 200)
+    cy.visit('http://localhost:3322/color/yellow')
+    cy.screenshot('overwrite-test', {
+      overwrite: false,
+    })
+
+    cy.screenshot('overwrite-test', {
+      overwrite: true,
+    })
+
+    cy.readFile(`cypress/screenshots/${path.basename(__filename)}/overwrite-test.png`).should('exist')
+    cy.readFile(`cypress/screenshots/${path.basename(__filename)}/overwrite-test (1).png`).should('not.exist')
```

Comment:
this test doesn't check if the screenshot was actually overwritten, only that an incremented filename was not created.

the test could be improved by checking that the bytes of `overwrite-test.png` are different after the new screenshot, by using `cy.readFile` to read the bytes of the screenshot png after each `cy.screenshot` call and comparing them.

you may need to pass `capture: 'runner'` to `cy.screenshot` so the bytes actually differ before/after the second `cy.screenshot`.